src/Decorator/Core/Content/Product/SalesChannel/Search/ProductSearchRoute.php

```diff
@@ -54,43 +55,43 @@ public function load(
     {
-        if (!SearchHelper::shouldHandleRequest($context, $this->configProvider)) {
-            return $this->decorated->load($request, $context, $criteria);
-        }
+        try {
+            if (!SearchHelper::shouldHandleRequest($context, $this->configProvider)) {
+                return $this->decorated->load($request, $context, $criteria);
+            }
 
-        if (!$request->get('search')) {
-            throw RoutingException::missingRequestParameter('search');
-        }
+            if (!$request->get('search')) {
+                throw RoutingException::missingRequestParameter('search');
+            }
 
-        if (!$request->get('order')) {
-            $request->request->set('order', ResolvedCriteriaProductSearchRoute::DEFAULT_SEARCH_SORT);
-        }
+            if (!$request->get('order')) {
+                $request->request->set('order', ResolvedCriteriaProductSearchRoute::DEFAULT_SEARCH_SORT);
+            }
 
-        $criteria->addState(Criteria::STATE_ELASTICSEARCH_AWARE);
+            $criteria->addState(Criteria::STATE_ELASTICSEARCH_AWARE);
 
-        $criteria->addFilter(
-            new ProductAvailableFilter(
-                $context->getSalesChannel()->getId(),
-                ProductVisibilityDefinition::VISIBILITY_SEARCH,
-            ),
-        );
+            $criteria->addFilter(
+                new ProductAvailableFilter(
+                    $context->getSalesChannel()->getId(),
+                    ProductVisibilityDefinition::VISIBILITY_SEARCH,
+                ),
+            );
 
-        $this->searchBuilder->build($request, $criteria, $context);
+            $this->searchBuilder->build($request, $criteria, $context);
 
-        $this->listingProcessor->prepare($request, $criteria, $context);
+            $this->listingProcessor->prepare($request, $criteria, $context);
 
-        $query = $request->query->get('search');
-        $result = $this->fetchProductsById($criteria, $context, $query);
-        $productListing = ProductListingResult::createFrom($result);
-        $this->eventDispatcher->dispatch(
-            new ProductSearchResultEvent($request, $productListing, $context),
-            ProductEvents::PRODUCT_SEARCH_RESULT
-        );
-        $productListing->addCurrentFilter('search', $query);
+            $query = $request->query->get('search');
+            $result = $this->fetchProductsById($criteria, $context, $query);
+            $productListing = ProductListingResult::createFrom($result);
+            $productListing->addCurrentFilter('search', $query);
 
-        $this->listingProcessor->process($request, $productListing, $context);
+            $this->listingProcessor->process($request, $productListing, $context);
 
-        return new ProductSearchRouteResponse($productListing);
+            return new ProductSearchRouteResponse($productListing);
+        } catch (Exception) {
+            return $this->decorated->load($request, $context, $criteria);
```

Comment:
Would be good to log this exception. If you know what can fail inside the block/what type of exception is being thrown, even better to catch separately and have more granular messages.

Reply:
I have added the logger